avax-synapse-stablepool/contracts/Strategy.sol

```diff
@@ -0,0 +1,483 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "../interfaces/curve.sol";
+import "../interfaces/IERC20Extended.sol";
+import "../interfaces/IWETH.sol";
+import "../interfaces/IUniswapV2Router02.sol";
+
+import "@tesrvaults/contracts/BaseStrategy.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+
+
+contract Strategy is BaseStrategy{
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    VaultAPI public yvToken;
+
+    IWETH public constant weth = IWETH(0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7);
+
+    address public constant router = address(0x1b02dA8Cb0d097eB8D57A175b88c7D8b47997506);
+
+    uint256 public lastInvest; // default is 0
+    uint256 public poolSize;
+    uint256 public minTimePerInvest;// = 3600;
+    uint256 public maxSingleInvest;// // 2 hbtc per hour default
+    uint256 public slippageProtectionIn;// = 50; //out of 10000. 50 = 0.5%
+    uint256 public slippageProtectionOut;// = 50; //out of 10000. 50 = 0.5%
+    uint256 public constant DENOMINATOR = 10_000;
+    string internal strategyName;
+
+
+    uint8 private want_decimals;
+
+    uint8 public curveId;
+    address public targetToken;
```

Comment:
Why do we have `targetToken` while the original strategy is using `want`? I think `want` comes from `BaseStrategy` and adding a duplicate variable could lead to errors in case the 2 vars get out of sync.